keras_retinanet/layers/nms.py

```diff
@@ -0,0 +1,189 @@
+"""
+Copyright 2017-2018 Fizyr (https://fizyr.com)
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import keras
+from .. import backend
+
+
+def nms(boxes, classification, other=[], score_threshold=0.05, max_detections=300, nms_threshold=0.5):
+    """ Run non maximum suppression using the boxes and classification values.
+
+    Args
+        boxes           : Tensor of shape (num_boxes, 4) containing the boxes in (x1, y1, x2, y2) format.
+        classification  : Tensor of shape (num_boxes, num_classes) containing the classification scores.
+        other           : List of tensors of shape (num_boxes, ...) to filter along with the boxes and classification scores.
+        score_threshold : Threshold used to prefilter the boxes with.
+        max_detections  : Maximum number of detections to filter.
+        nms_threshold   : Threshold for the IoU value to determine when a box should be suppressed.
+
+    Returns
+        A list of [boxes, scores, labels, other[0], other[1], ...].
+        boxes is shaped (max_detections, 4) and contains the (x1, y1, x2, y2) boxes of the non-suppressed boxes.
+        scores is shaped (max_detections,) and contains the scores of the predicted class.
+        labels is shaped (max_detections,) and contains the predicted label.
+        other[i] is shaped (max_detections, ...) and contains the filtered other[i] data.
+        In case there are less than max_detections detections, the tensors are padded with -1's.
+    """
+    all_boxes  = []
+    all_scores = []
+    all_labels = []
+    all_other  = []
+
+    # perform per class NMS
+    for c in range(int(classification.shape[1])):
```

Comment:
Aren't shapes always ints anyway?

Reply:
They're actually tensors, but in this case they are always representing an int.

Reply:
I did say int**s** (plural) ^^ But I'm wondering if the cast to `int` is really needed.